Cppcheck Charlie 30

GuiMacielPereira opened the pull request:
### Description of work

#### Summary of work
<!-- Please provide a short, high level description of the work that was done.
-->

<!-- Why has this work been done? If there is no linked issue please provide appropriate context for this work.
#### Purpose of work
This can be removed if a github issue is referenced below
-->

Fixes cppcheck set http://172.16.111.171/suppressionSet.php?setid=30. <!-- and fix #xxxx or close #xxxx xor resolves #xxxx. One line per issue fixed. -->
<!-- alternative
*There is no associated issue.*
-->

<!-- If the original issue was raised by a user they should be named here. Do not leak email addresses
**Report to:** [user name]
-->

#### Further detail of work
<!-- Please provide a more detailed description of the work that has been undertaken.
-->

### To test:

<!-- Instructions for testing.
There should be sufficient instructions for someone unfamiliar with the application to test - unless a specific
reviewer is requested.
If instructions for replicating the fault are contained in the linked issue then it is OK to refer back to these.
-->

<!-- delete this if you added release notes
*This does not require release notes* because **fill in an explanation of why**
If you add release notes please save them as a separate file using the Issue or PR number as the file name. Check the file is located in the correct directory for your note(s).
-->

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

GuiMacielPereira committed:
Delete suppresions for set 30

GuiMacielPereira committed:
missingOverride line 84

GuiMacielPereira committed:
Fix uninitMemberVarPrivate

I think it was complaining because the variable was not initialized outside the if/else statements.

GuiMacielPereira committed:
False positive with uselessOverride

GuiMacielPereira committed:
Fix missingOverride

GuiMacielPereira committed:
Fix missing const and false positive

GuiMacielPereira committed:
Fix containerOutOfBounds

Cppcheck complained that polynomialCoefficients is accessed when empty.
I think this change to the code is valie because the size of polynomialCoefficients
should be the same as the size of labels.
Also since the main goal of the loop is to run through the coefficients, it makes the intent clearer.

GuiMacielPereira committed:
Fix constVariableReference

For loop does not change ws, it only adds it to wsg

GuiMacielPereira committed:
Fix constVariablePointer

This pointer is not used to change the data it's pointing to.

GuiMacielPereira committed:
Fix duplicateConditionalAssign

This stement is just reassigning is_symbol to it's own value.

GuiMacielPereira committed:
Fix constVariableReference

GuiMacielPereira committed:
Fix cstyleCast

GuiMacielPereira committed:
Fix uninitMemberVarPrivate (again)

My last change did not fix it because the variables were not initialized in the header.

GuiMacielPereira committed:
Remove my previous suppression with better solution

Using pop_back() in this case is the perfect solution, since the whole point is the remove a trailing comma.

GuiMacielPereira committed:
Fix uninitMemberVarPrivate

Initialize to 0.0 because there is a check that already assumes this is the value when the user has not correctly set it.

GuiMacielPereira committed:
Revert "Fix uninitMemberVarPrivate"

This reverts commit 746ffaf6481b96f2e43fbd287ef05e4c643038ea.

I am reverting this change because this causes unmatched cppcheck
suppressions that might cause merge conflicts, so I would rather not
touch this file for now.

GuiMacielPereira committed:
Fix new cppcheck warnings

Got new warnings on how the variable was not being used, so I removed it.

GuiMacielPereira committed:
Fix unsignedLessThanZero due to my previous changes

Got warning due to using a size_t variable inside the comparison in the for loop.
Fixed it by changing it ot a sigend size_t.

GuiMacielPereira commented:
I know I am so sorry 😭 it's so messy as well

sf1919 reviewed (dismissed):
Looks good to me. Took me a while to work out the is_symbol stuff in FindDetectorsPar but I do think it probably should be removed altogether.

Ready for @jclarkeSTFC to gatekeep

Team Bravo reviewing

sf1919 commented on Framework/DataHandling/src/CreatePolarizationEfficiencies.cpp:
```diff
@@ -114,7 +114,7 @@ MatrixWorkspace_sptr CreatePolarizationEfficiencies::createEfficiencies(std::vec
   auto const x = inWS->points(0);
   std::vector<double> y(x.size());
-  for (size_t i = 0; i < labels.size(); ++i) {
+  for (ssize_t i = 0; i < polynomialCoefficients.size(); ++i) {
```
Looks like this one is causing problems with tests

GuiMacielPereira committed:
Fix ssize_t

Did not realize this type was unique to Unix.

GuiMacielPereira committed:
Initialize polynomialCoefficients

This will hopefully avoid Cppcheck complaining about size_t being compared to be less than zero.

GuiMacielPereira committed:
Use transform instead of for loop

GuiMacielPereira committed:
Fix compiler error

Compiler was complaining because I used cbegin() instead of begin()

GuiMacielPereira committed:
Better transform using ranges

GuiMacielPereira committed:
include ranges

jclarkeSTFC commented on Framework/DataHandling/src/LoadANSTOHelper.cpp:
```diff
@@ -207,7 +207,8 @@ namespace Tar {
   memset(Checksum, ' ', sizeof(Checksum));
-  size_t value = std::accumulate((const char *)this, (const char *)this + sizeof(EntryHeader), (size_t)0);
+  size_t value = std::accumulate(reinterpret_cast<const char *>(this),
+                                 reinterpret_cast<const char *>(this) + sizeof(EntryHeader), static_cast<size_t>(0));
```
```suggestion
                                 reinterpret_cast<const char *>(this) + sizeof(EntryHeader), size_t(0));
```

GuiMacielPereira replied:
I tried that but cppcheck complains that it's still cstyle

jclarkeSTFC reviewed (changes requested):
Looks good, just one small thing

Reviewed by Bravo

GuiMacielPereira committed:
Remove wrong semantics

GuiMacielPereira committed:
Removed ranges

jclarkeSTFC reviewed (approved):
👍

sf1919 reviewed (approved):
Thanks for continuing to work on this one. Hopefully it's resolved the failing test issue!

sf1919 commented:
Not convinced by linux packages build fail so have set it to re-run but it's currently queued.
